error_middleware.go

```diff
@@ -11,14 +11,15 @@ import (
 	"github.com/sirupsen/logrus"
 	"github.com/urfave/negroni"
 
-	"github.com/Scalingo/go-utils/errors"
+	"github.com/Scalingo/go-utils/errors/v2"
 	"github.com/Scalingo/go-utils/logger"
 	"github.com/Scalingo/go-utils/security"
 )
 
 var ErrorMiddleware = MiddlewareFunc(func(handler HandlerFunc) HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request, vars map[string]string) error {
-		log, ok := r.Context().Value("logger").(logrus.FieldLogger)
+		ctx := r.Context()
+		log, ok := ctx.Value("logger").(logrus.FieldLogger)
```

Comment:
question: that may be a silly question but why this and not `logger.Get(ctx)`?

Reply:
Because of the difference of behavior related to the presence of "logger" value or not.
Here, if there is no logger in the context we create a new one.
In `logger.Get` we use Default and `invalid_context`.

But after checking on: https://github.com/Scalingo/go-handlers/blob/e6aa76bc2226a34db4880ad7815d333e1489c396/logging_middleware.go#L76
the logger is always set by `Apply`. So I think we can use `logger.Get` indeed :+1:

Reply:
Mmmmh but if we use `logger.Get` it means that we make it mandatory for the logging middleware to be loaded _before_ the error middleware,  don't we?

I'm not sure we want to enforce the middleware order and with your explanation I think we should keep the previous behaviour. What do you think?

Reply:
> Mmmmh but if we use logger.Get it means that we make it mandatory for the logging middleware to be loaded before the error middleware, don't we?

For my comprehension. It makes the usage of `NewRouter` mandatory because this function injects the logger in the context.
cf: https://github.com/Scalingo/go-handlers/blob/e6aa76bc2226a34db4880ad7815d333e1489c396/router.go#L13-L19

Or it should be manually set like it is done for example in https://github.com/Scalingo/metrics-storage-service/blob/b332d454c0a9662fde549e54bd515f1a2f2014f8/webserver/webserver.go#L23-L26

Reply:
To avoid any disruption in services that implement it. I will keep the first implementation.